src/module/__tests__/helpers-dice.test.ts

```diff
@@ -0,0 +1,484 @@
+        const roll = createMockRoll(10);
+        expect(OseDice.digestResult(data, roll).isSuccess).equal(true);
+        expect(OseDice.digestResult(data, roll).isFailure).equal(false);
+      });
+      it("Successful roll under", () => {
+        const roll = createMockRoll(9);
+        expect(OseDice.digestResult(data, roll).isSuccess).equal(true);
+        expect(OseDice.digestResult(data, roll).isFailure).equal(false);
+      });
+      it("Unsuccessful roll over", () => {
+        const roll = createMockRoll(11);
+        expect(OseDice.digestResult(data, roll).isSuccess).equal(false);
+        expect(OseDice.digestResult(data, roll).isFailure).equal(true);
+      });
+    });
+    describe("table type", () => {
+      const data = {
+        roll: {
+          type: "table",
+          target: 10,
+          table: {
+            9: "Success",
+            10: "Failure",
+          },
+        },
+      };
+
+      it("Successful roll hitting target", () => {
+        const roll = createMockRoll(9);
+        expect(OseDice.digestResult(data, roll).isSuccess).equal(false);
+        expect(OseDice.digestResult(data, roll).isSuccess).equal(false);
+        expect(OseDice.digestResult(data, roll).details).equal("Success");
+      });
+      it("Successful roll under", () => {
+        const roll = createMockRoll(10);
+        expect(OseDice.digestResult(data, roll).isSuccess).equal(false);
+        expect(OseDice.digestResult(data, roll).isSuccess).equal(false);
+        expect(OseDice.digestResult(data, roll).details).equal("Failure");
+      });
+    });
+    describe("unknown type", () => {
+      const data = createMockData("unknown", 10);
+      it("Returns default result", () => {
+        const roll = createMockRoll(4);
+        expect(OseDice.digestResult(data, roll).isSuccess).equal(false);
+        expect(OseDice.digestResult(data, roll).isSuccess).equal(false);
+        expect(OseDice.digestResult(data, roll).target).equal(10);
+        expect(OseDice.digestResult(data, roll).total).equal(4);
+      });
+    });
+  });
+  describe("attackIsSuccess(roll, thac0, ac)", () => {
+    it("Natural 1 total always fails", () => {
+      const roll = createMockRoll(1);
+      expect(OseDice.attackIsSuccess(roll, 10, 10)).equal(false);
+    });
+    it("Natural 1 terms always fails", () => {
+      const roll = createMockRoll(5, [1]);
+      expect(OseDice.attackIsSuccess(roll, 10, 10)).equal(false);
+    });
+    it("Natural 20 total always succeed", () => {
+      const roll = createMockRoll(20);
+      expect(OseDice.attackIsSuccess(roll, 100, 10)).equal(true);
+    });
+    it("Natural 20+ total always succeed", () => {
+      const roll = createMockRoll(21);
+      expect(OseDice.attackIsSuccess(roll, 100, 10)).equal(true);
+    });
+    it("Natural 20 terms always succeed", () => {
+      const roll = createMockRoll(5, [20]);
+      expect(OseDice.attackIsSuccess(roll, 100, 10)).equal(true);
```

Comment:
As mentioned in Discord and logged in #340, please do something like

```js
createMockRoll(1,[1])
expect(OseDice.attackIsSuccess(roll, 21, 0)).equal(false);
createMockRoll(1,[20])
expect(OseDice.attackIsSuccess(roll, 0, 21)).equal(true);
createMockRoll(20,[1])
expect(OseDice.attackIsSuccess(roll, 21, 0)).equal(false);
createMockRoll(20,[20])
expect(OseDice.attackIsSuccess(roll, 0, 21)).equal(true);
```

(I'm only 50% sure those are the values we need in attackIsSuccess)

Reply:
Updated the tests with only these for testing natural 1 and 20s!